lib/vimgolf/cli.rb

```diff
@@ -61,43 +60,69 @@ def setup
       end
     end
 
-    desc "put [ID]", "launch Vim session"
+    desc "put CHALLENGE_ID", "launch vimgolf.com challenge"
     long_desc <<-DESC
     Launch a VimGolf session for the specified challenge ID. To find a currently
     active challenge ID, please visit vimgolf.com!
     DESC
 
-    def put(id = nil)
+    def put(id)
       FileUtils.mkdir_p Config.put_path
       VimGolf.ui.warn "Downloading Vimgolf challenge: #{id}"
       VimGolf::Challenge.path(Config.put_path)
       challenge = Challenge.new(id)
       challenge.download
 
+      play(challenge)
+    end
+
+    desc "local INFILE OUTFILE", "launch local challenge"
+    long_desc <<-DESC
+    Launch a local VimGolf challenge. A temporary copy of INFILE is made; the original files will not be touched.
+    DESC
+
+    def local(infile, outfile)
+      # make sure our files are sane
+      if !(File.file?(infile) and File.file?(outfile))
```

Comment:
I thought of that before I commited. If you `vimgolf local foo.txt foo.txt` it behaves as if it's running on separate files that are identical. It's not a bad behavior, so I did nothing about it.